crates/runtime/src/trampolines.rs

```diff
@@ -0,0 +1,43 @@
+//! Trampolines for calling into Wasm from the host and calling the host from
+//! Wasm.
+
+use crate::VMContext;
+use std::mem;
+
+/// Given a Wasm function pointer and a `vmctx`, prepare the `vmctx` for calling
+/// into that Wasm function, and return the host-to-Wasm entry trampoline.
+///
+/// Callers must never call Wasm function pointers directly. Callers must
+/// instead call this function and then enter Wasm through the returned
+/// host-to-Wasm trampoline.
+#[inline]
```

Comment:
Could you add some notes to the docs here about the unsafety and how `T` must be carefully selected?